### PR TITLE
libkb: use correct convention for revoked last-writer-wins chainlinks

### PR DESCRIPTION
**Pull request by maxtaco**
Before if we had:

   bob@twitter -> alice@twitter -> revoke(alice@twitter)

We'd still check bob@twitter.  I think that's wrong, so fix it with this
commit

### CODE REVIEW

**Comment by maxtaco** [comment]
@oconnor663 PTAL?

File: go/service/main.go [hunk 1]
```diff
@@ -148,7 +148,10 @@ func (d *Service) Run() (err error) {
 		}
 	}
 
-	// Explicitly set fork type here based on KEYBASE_LABEL
+	// Explicitly set fork type here based on KEYBASE_LABEL.
+	// This is for OSX-based Launchd implementations, which unfortunately
+	// don't obey the same command-line flag conventions as
+	// the other platforms.
```
Comment (oconnor663) [inline go/service/main.go]: Unrelated change snuck in?

File: go/libkb/id_table.go [hunk 2]
```diff
@@ -1039,15 +1039,17 @@ func NewIdentityTable(g *GlobalContext, eldest keybase1.KID, sc *SigChain, h *Si
...
 
-func (idt *IdentityTable) populate() error {
-	G.Log.Debug("+ Populate ID Table")
-	links, err := idt.sigChain.GetCurrentSubchain(idt.eldest)
-	if err != nil {
+func (idt *IdentityTable) populate() (err error) {
+	defer idt.G().Trace("IdentityTable::populate", func() error { return err })()
+
+	var links []*ChainLink
+	if links, err = idt.sigChain.GetCurrentSubchain(idt.eldest); err != nil {
```
Comment (oconnor663) [inline go/libkb/id_table.go]: The style war continues :-D

File: go/libkb/remote_proof_links.go [hunk 3]
```diff
@@ -139,25 +139,20 @@ func AddToProofSetNoChecks(r RemoteProofChainLink, ps *ProofSet) {
...
-			id := link.ToDisplayString()
-			if seen[id] {
-				continue
+			if !link.IsRevoked() {
+				id := link.ToDisplayString()
+				// We only want to use the last proof in the list
+				// if we have several (like for dns://chriscoyne.com)
+				if _, ok := seen[id]; !ok {
+					links = append(links, both)
+					seen[id] = struct{}{}
```
Comment (oconnor663) [inline go/libkb/remote_proof_links.go]: Are `id` strings guaranteed to be canonical? If I have one proof for "jack@twitter" and another proof for "JACK@twitter", is it possible this check will get confused?

Comment (oconnor663) [inline go/libkb/remote_proof_links.go]: Also, I thought we were going to allow "anothername@twitter" to implicitly revoke "previousname@twitter", but here it looks like it won't?

Comment (maxtaco) [inline go/libkb/remote_proof_links.go]: hmm, maybe not, good point. it's confusing though, since with HN at least, names are case-sensitive, but you can't have 2 usernames that are the same if toLower'ed.

Comment (maxtaco) [inline go/libkb/remote_proof_links.go]: with this new change, you can only skip in the else of 151 if you are not `LastWriterWins`.

**Comment by oconnor663** [comment]
Do we have a task for the corresponding libkeybase-js changes?

**Comment by oconnor663** [comment]
:shipit: 

Recording my thoughts for future-Jack:

On the one hand, having signatures interact with each other like this makes sigchain parsing more complicated, because the checker can't just drop revoked links from history. On the other hand, having recent identities always suppress old ones makes a correctly-implemented checker more conservative, which is a good thing. So on balance, I'm for it.
